### ISSUE TITLE
Home function on Instruments requires multiple steps, shutter

### ISSUE THREAD

**Issue by kameyer226** [yaq-project/yaqc-qtpy#14]
Home button requires pressing twice to ensure motors actually home.   The shutter closes for the homing procedure but the qtpy indicator for that shutter closing does not update.

**Comment by ksunden** [comment]
There are two separate issues indicated here:


First, the shutter behavior is controlled by the lightcon software, we have no good way of knowing that the state changed.


As to the needing to press home multiple times:

I'm skeptical about the "requires pressing twice" comment, my instinct is that that is due to lack of patience rather than lack of initiating a home.

Which specific hardware does this apply to?

Can you replicate the behavior using the python api rather than the UI (if so, it is a daemon bug, not a qtpy bug)

**Comment by untzag** [comment]
I agree with @ksunden, could you try this with other hardware or with the Python API? Perhaps the underlying problem here is there isn't good feedback of "I'm homing now"? Does the hardware immediately go busy when you click home?

Thanks for raising this issue @kameyer226

**Comment by kameyer226** [comment]
I can hear motors moving the second time home is pressed but not the first time.    Second issue:   since home always closes the shutter why not let qtpy know that it is closed that way?

**Comment by ksunden** [comment]
> Second issue: since home always closes the shutter why not let qtpy know that it is closed that way?

As I already stated, we _do not_ have access to fact that that the shutter state changes for this reason. that is why not, there is zero interaction with the shutter daemon.

**Comment by untzag** [comment]
@kameyer226 could you engage with some of our questions above please?

What is the behavior for other hardware?

What is the behavior using the Python API?

As you know, there are many layers to our software stack. Some clarification will help us define exactly which piece of software is behaving in unexpected ways.

**Comment by kameyer226** [comment]
When home is pressed for our OPAs,  the shutters close automatically, but qtpy does not show they are closed.   Thus qtpy is misreporting the shutter condition after the home which is unsafe for the users.

**Comment by untzag** [comment]
@kameyer226 it seems like the questions I asked above are of no interest to you. This must mean that you already fully understand the problem. What changes to yaqc-qtpy would you like to make? Since I am an idiot, it would be helpful if you also explained very slowly exactly how those proposed changes would solve the problem you report.

**Comment by kameyer226** [comment]
The other hardware doesn't have shutter features associated with them.   For the home "double-clicking" features, we are in the middle of runs now to be able to diagnose whether we have to press home twice there, and so there are some delays in responding to this.

**Comment by ddkohler** [comment]
It sounds like the "OPOA daemon does not report on shutter state when homing" issue is not in the scope of yaqc-qtpy.  @kameyer226 you could raise an issue for lightcon daemon, but it doesn't seem like we have a great fix at this time.  We just don't get robust shutter info from lightcon. We just have to accept that daemon shutter state should not be trusted absolutely, and users should use other means to guarantee that the light is contained when they go onto the laser table.  If the concern is safety, it's proper to interface the table with the explicit interlocks provided by the laser.  

As for "home button requires two presses", it's worth diagnosing further.  @kameyer226, why do you think this is a yaqc-qtpy issue?  e.g. when you use a yaqc client and call the home command, do you need to call it twice?  That's pretty much all yaqc-qtpy is doing on the button press, so it would be surprising if problem lies in yaqc-qtpy.  @kameyer226 more information on whether the "home" double-click issue happens with other daemons would also clue us into whether the problem is in yaqc-qtpy.

**Comment by kameyer226** [comment]
I will discount the shutter issue when homing until I can read the positions vs Client, but retain the double-clicking aspect of home until I can test the other controls out to see what they do when homing.
